File: frontend/src/api/me.ts
export const ME_PATH = "/api/v1/me";

export type ApplicationRole = "user" | "owner";

export type MeResponse = {
  application_role: ApplicationRole;
  is_owner: boolean;
};

export function parseMeResponse(payload: unknown): MeResponse | null {
  if (typeof payload !== "object" || payload === null) {
    return null;
  }
  const body = payload as Record<string, unknown>;
  const role = body.application_role;
  const isOwner = body.is_owner;
  if (role !== "user" && role !== "owner") {
    return null;
  }
  if (typeof isOwner !== "boolean") {
    return null;
  }
  if (isOwner !== (role === "owner")) {
    return null;
  }
  return { application_role: role, is_owner: isOwner };
}

File: app/schemas/me.py
"""Response schemas for the current-user identity endpoint."""

from typing import Literal

from pydantic import BaseModel, Field


class MeResponse(BaseModel):
    """Minimal server-authoritative application identity for the caller.

    Omits internal ``user_id``: the SPA needs only role for presentation.
    Never includes issuer, subject, email, mailbox identity, or tokens.
    """

    application_role: Literal["user", "owner"] = Field(
        description="Persisted ECI application_role for the resolved user.",
        examples=["user"],
    )
    is_owner: bool = Field(
        description="True only when application_role is owner. UX hint only.",
        examples=[False],
    )
